app/src/main/java/com/swent/assos/ui/screens/assoDetails/EventDetails.kt

```diff
@@ -57,113 +53,104 @@ fun EventDetails(eventId: String, navigationActions: NavigationActions, assoId:
-                        } else {
-                          eventViewModel.applyStaffing(
-                              event.id,
-                              userId.id,
-                              Toast.makeText(
-                                      context,
-                                      "You have successfully applied to staff at the event",
-                                      Toast.LENGTH_SHORT)
-                                  .show())
-                        }
-                      },
-                      label = {
-                        if (applied.value) {
-                          Text(
-                              text = "Remove staff application",
-                              color = MaterialTheme.colorScheme.onSurfaceVariant,
-                              fontFamily = FontFamily(Font(R.font.sf_pro_display_regular)),
-                              fontWeight = FontWeight.Medium,
-                          )
-                        } else {
-                          Text(
-                              text = "Apply for staffing",
-                              color = MaterialTheme.colorScheme.onSecondary,
-                              fontFamily = FontFamily(Font(R.font.sf_pro_display_regular)),
-                              fontWeight = FontWeight.Medium,
-                          )
-                        }
-                      },
-                  )
-                }
-          }
+        if (!(associations.map { it.id }.contains(assoId)) &&
+            event.isStaffingEnabled &&
+            !loading.value) {
+          AssistChip(
+              colors = staffButtonColors(),
+              border = null,
+              modifier = Modifier.testTag("StaffButton").padding(5.dp),
+              onClick = { callBackStaffRequest() },
+              label = { labelStaffButton() },
+          )
         }
       },
       floatingActionButtonPosition = FabPosition.Center) { paddingValues ->
-        if (conf) {
-          ConfirmDialog(
-              onDismiss = { conf = false },
-              onConfirm = {
-                conf = false
-                eventViewModel.deleteEvent(eventId)
-                navigationActions.goBack()
-              },
-              title = event.title)
+        if (loading.value) {
+          LoadingCircle()
+        } else {
+          if (conf) {
+            ConfirmDialog(
+                onDismiss = { conf = false },
+                onConfirm = {
+                  conf = false
+                  eventViewModel.deleteEvent(eventId)
+                  navigationActions.goBack()
+                },
+                title = event.title)
+          }
+          EventContent(
+              viewModel = eventViewModel,
+              paddingValues = paddingValues,
+              isMember = (associations.map { it.id }.contains(assoId)),
+              eventId = eventId,
+              navigationActions = navigationActions)
```

Comment:
a nested if can result in a lot of complexity is it something you thought about ?

Reply:
yes you are right I could put it out